use-cases/ai-agent.ipynb

```diff
@@ -0,0 +1,560 @@
+    "    \"fields\": [\n",
+    "      {\n",
+    "        \"type\": \"vector\",\n",
+    "        \"numDimensions\": 1024,\n",
+    "        \"path\": \"embedding\",\n",
+    "        \"similarity\": \"cosine\"\n",
+    "      }\n",
+    "    ]\n",
+    "  },\n",
+    "  name = index_name,\n",
+    "  type = \"vectorSearch\"\n",
+    ")\n",
+    "collection.create_search_index(model=search_index_model)\n",
+    "\n",
+    "# Wait for initial sync to complete\n",
+    "print(\"Polling to check if the index is ready. This may take up to a minute.\")\n",
+    "predicate=None\n",
+    "if predicate is None:\n",
+    "   predicate = lambda index: index.get(\"queryable\") is True\n",
+    "\n",
+    "while True:\n",
+    "   indices = list(collection.list_search_indexes(index_name))\n",
+    "   if len(indices) and predicate(indices[0]):\n",
+    "      break\n",
+    "   time.sleep(5)\n",
+    "print(index_name + \" is ready for querying.\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "pZfheX5FiIhU"
+   },
+   "source": [
+    "## Define tools for the agent\n",
+    "Next, you define two tools that the agent can use to complete tasks:\n",
+    "- `vector_search_tool`: Runs a vector search query to retrieve semantically similar documents from Atlas.\n",
+    "- `calculator_tool`: Uses the `eval()` function which can be used for math operations on a string. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define a vector search tool\n",
+    "def vector_search_tool(user_input: str) -> str:\n",
+    "    query_embedding = get_embedding(user_input)\n",
+    "    pipeline = [\n",
+    "        {\n",
+    "              \"$vectorSearch\": {\n",
+    "                \"index\": \"vector_index\",\n",
+    "                \"queryVector\": query_embedding,\n",
+    "                \"path\": \"embedding\",\n",
+    "                \"exact\": True,\n",
+    "                \"limit\": 5\n",
+    "              }\n",
+    "        }, {\n",
+    "              \"$project\": {\n",
+    "                \"_id\": 0,\n",
+    "                \"text\": 1\n",
+    "           }\n",
+    "        }\n",
+    "    ]\n",
+    "\n",
+    "    results = collection.aggregate(pipeline)\n",
+    "\n",
+    "    array_of_results = []\n",
+    "    for doc in results:\n",
+    "        array_of_results.append(doc)\n",
+    "    return array_of_results"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define a simple calculator tool\n",
+    "def calculator_tool(user_input: str) -> str:\n",
+    "    try:\n",
+    "        result = eval(user_input)\n",
+    "        return str(result)\n",
+    "    except Exception as e:\n",
+    "        return f\"Error: {str(e)}\" "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
```

Comment:
Just checking - do we _intend_ to provide the output here, or did this get accidentally included from testing? Usually we have omitted any output and let users execute the cells to see the output.